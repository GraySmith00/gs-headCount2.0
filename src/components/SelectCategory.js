import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SelectCategory extends Component {
  constructor() {
    super();
    this.state = {
      value: ''
    };
  }

  handleChange = e => {
    this.setState({
      value: e.target.value
    });
    this.props.changeCategory(e.target.value);
  };

  render() {
    const categories = [
      'Kindergartners In Full Day Propgram',
      'Third Grade Tests',
      'Eigth Grade Test Scores',
      'Average Race Ethnicity MathScores'
    ];
    let options = categories.map((category, i) => (
      <option key={`category${i}`} value={i}>
        {category}
      </option>
    ));
    return (
      <div>
        <select
          className="select-category"
          value={this.state.value}
          onChange={this.handleChange}
        >
          {options}
        </select>
      </div>
    );
  }
}

SelectCategory.propTypes = {
  changeCategory: PropTypes.func.isRequired
};

export default SelectCategory;
