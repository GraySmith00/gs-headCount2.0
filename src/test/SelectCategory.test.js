import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import SelectCategory from '../components/SelectCategory';

describe('SelectCategory component', () => {
  let wrapper;
  let changeCategory;

  beforeEach(() => {
    changeCategory = jest.fn();
    wrapper = shallow(<SelectCategory changeCategory={changeCategory} />);
  });

  it('renders without crashing', () => {
    const changeCategory = jest.fn();
    const div = document.createElement('div');
    ReactDOM.render(<SelectCategory changeCategory={changeCategory} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should have an initail state', () => {
    expect(wrapper.state().value).toEqual('');
  });

  it('should change the value state when changing the select value', () => {
    wrapper.find('.select-category').simulate('change', {
      target: { value: 'Kindergartners In Full Day Propgram' }
    });

    expect(wrapper.state().value).toEqual(
      'Kindergartners In Full Day Propgram'
    );
  });

  it('should call the changeCategory method', () => {
    wrapper.find('.select-category').simulate('change', {
      target: { value: 'Kindergartners In Full Day Propgram' }
    });

    expect(changeCategory).toHaveBeenCalled();
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
