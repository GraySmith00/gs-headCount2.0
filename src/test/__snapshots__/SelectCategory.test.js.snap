// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectCategory component should match the snapshot 1`] = `"<div><select class=\\"select-category\\"><option value=\\"0\\">Kindergartners In Full Day Propgram</option><option value=\\"1\\">Third Grade Tests</option><option value=\\"2\\">Eigth Grade Test Scores</option><option value=\\"3\\">Average Race Ethnicity MathScores</option></select></div>"`;
